dev.goats.xivlauncher.yml

```diff
@@ -0,0 +1,70 @@
+app-id: dev.goats.xivlauncher
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.dotnet6
+build-options:
+  build-args:
+    - --share=network
+  append-path: "/usr/lib/sdk/dotnet6/bin"
+  append-ld-library-path: "/usr/lib/sdk/dotnet6/lib"
+  env:
+    PKG_CONFIG_PATH: "/app/lib/pkgconfig:/app/share/pkgconfig:/usr/lib/pkgconfig:/usr/share/pkgconfig:/usr/lib/sdk/dotnet6/lib/pkgconfig"
+    DOTNET_CLI_TELEMETRY_OPTOUT: "true"
+command: /app/opt/XIVLauncher/XIVLauncher.Core
+modules:
+  - name: libsecret
+    buildsystem: meson
+    config-opts:
+      - "-Dmanpage=false"
+      - "-Dvapi=false"
+      - "-Dgtk_doc=false"
+    cleanup:
+      - /bin
+      - /include
+      - /lib/pkgconfig
+      - /share/gir-1.0
+      - /share/man
+    sources:
+      - type: archive
+        url: https://download.gnome.org/sources/libsecret/0.20/libsecret-0.20.5.tar.xz
+        sha256: 3fb3ce340fcd7db54d87c893e69bfc2b1f6e4d4b279065ffe66dac9f0fd12b4d
+  - name: aria2
+    config-opts:
+      - '--with-ca-bundle=/etc/ssl/certs/ca-certificates.crt'
+    sources:
+      - type: archive
+        url: https://github.com/aria2/aria2/releases/download/release-1.36.0/aria2-1.36.0.tar.xz
+        sha256: 58d1e7608c12404f0229a3d9a4953d0d00c18040504498b483305bcb3de907a5
+  - name: xivlauncher
+    buildsystem: simple
+    build-commands:
+      - install -d /app/opt/XIVLauncher/
+      - dotnet publish -r linux-x64 --sc -o /app/opt/XIVLauncher/ /run/build/xivlauncher/src/XIVLauncher.Core/XIVLauncher.Core.csproj --configuration Release
+      - install -D -m644 /run/build/xivlauncher/XIVLauncher.desktop /app/share/applications/dev.goats.xivlauncher.desktop
+      - install -D -m644 /run/build/xivlauncher/512.png /app/share/icons/hicolor/512x512/apps/dev.goats.xivlauncher.png
+      - install -D -m644 /run/build/xivlauncher/dev.goats.xivlauncher.appdata.xml /app/share/metainfo/dev.goats.xivlauncher.appdata.xml
+    sources:
+      - type: git
+        url: https://github.com/goatcorp/FFXIVQuickLauncher.git
+        commit: 473cdef17c13792133abc7efee8417255294e448
+      - type: file
+        path: XIVLauncher.desktop
+      - type: file
+        path: 512.png
+      - type: file
+        path: dev.goats.xivlauncher.appdata.xml
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --share=network
+  - --filesystem=home
+  - --device=dri
+  - --socket=pulseaudio
+  - --talk-name=org.freedesktop.secrets
+  - --system-talk-name=org.freedesktop.UDisks2
+  - --device=all
+  - --allow=devel
```

Comment:
why do you need `devel`?

Reply:
Devel is needed to inject a plugin system into the game which is a feature of this launcher.

Reply:
can you explain?

Reply:
The application we run in WINE injects into a process also running in the same WINE instance - the Windows API functions used for this are implemented in wine using self-debugging techniques and syscalls on Linux.

https://github.com/wine-mirror/wine/blob/1d178982ae5a73b18f367026c8689b56789c39fd/dlls/kernelbase/debug.c#L929-L995 (implementation of one of the functions we are calling - the module list won't be populated if debugging isn't available, and ERROR_NOACCESS is indicated by the function)